### PR TITLE
Design form Course

### PR DESCRIPTION
**Pull request by phornsopheak**

### CODE REVIEW

File: app/views/supervisor/courses/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<% provide(:title, "Create course") %>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: remove `()`

File: app/views/supervisor/courses/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      <%= f.label :name %>
+      <%= f.text_field :name, class: "form-control" %>
+
+      <%= f.label :description %>
+      <%= f.text_area :description, class: "form-control" %>
+
+      <div class="container-fluid">
+      <%= f.label :status %>
+      <div class="row">
+      <div class="col-md-1"> <%= f.label :status, "Yes", :value => "true" %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: coding convention

File: app/views/supervisor/courses/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      <div class="col-md-1"> <%= f.radio_button :status, "start" %> </div>
+      </div>
+      <div class="row">
+      <div class="col-md-1"> <%= f.label :status, "No", :value => "false" %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: coding convention

File: app/views/supervisor/courses/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      <div class="col-md-1"> <%= f.radio_button :status, "not start", :checked => true %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: coding convention

File: app/views/shared/_error_messages.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(object.errors.count, "error") %>.
+    </div>
+    <ul>
+    <% object.errors.full_messages.each do |msg| %>
+      <li><%= msg %></li>
+    <% end %>
+    </ul>
+  </div>
+<% end %>
```
Comment (samnangcattor) [inline app/views/shared/_error_messages.html.erb]: new line  in the end

File: app/views/supervisor/courses/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      </div>
+      </div>
+      <%= f.submit "Create Course", class: "btn btn-primary" %>
+    <% end %>
+  </div>
+</div>
```
Comment (samnangcattor) [inline app/views/supervisor/courses/new.html.erb]: new line in the end file

**Comment by samnangcattor** [comment]
ok

File: app/views/supervisor/courses/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    <%= form_for [:supervisor, @course] do |f| %>
+      <%= render "shared/error_messages", object: f.object %>
+      <%= f.label :name %>
+      <%= f.text_field :name, class: "form-control" %>
+      <%= f.label :description %>
+      <%= f.text_area :description, class: "form-control" %>
+      <div class="container-fluid">
+       <%= f.label :status %>
+       <div class="row">
+        <div class="col-md-1"> <%= f.label :status, "Yes", :value => "true" %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: `:value => "true"` => `value: true`

File: app/views/supervisor/courses/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        <div class="col-md-1"> <%= f.radio_button :status, "start" %> </div>
+      </div>
+      <div class="row">
+       <div class="col-md-1"> <%= f.label :status, "No", :value => "false" %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: `:value => "true"` => `value: false`

File: app/views/supervisor/courses/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+       <div class="col-md-1"> <%= f.radio_button :status, "not start", :checked => true %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: `:checked => true` => `checked: true `

File: app/views/supervisor/courses/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+      <%= f.label :description %>
+      <%= f.text_area :description, class: "form-control" %>
+      <div class="container-fluid">
+       <%= f.label :status %>
+       <div class="row">
+        <div class="col-md-1"> <%= f.label :status, "Yes", :value => true %> </div>
+        <div class="col-md-1"> <%= f.radio_button :status, "start" %> </div>
+      </div>
+      <div class="row">
+       <div class="col-md-1"> <%= f.label :status, "No", :value => false %> </div>
```
Comment (RathanakSreang) [inline app/views/supervisor/courses/new.html.erb]: `value: true`

File: app/views/supervisor/courses/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    <%= form_for [:supervisor, @course] do |f| %>
+      <%= render "shared/error_messages", object: f.object %>
+      <%= f.label :name %>
+      <%= f.text_field :name, class: "form-control" %>
+      <%= f.label :description %>
+      <%= f.text_area :description, class: "form-control" %>
+      <div class="container-fluid">
+       <%= f.label :status %>
+       <div class="row">
+        <div class="col-md-1"> <%= f.label :status, "Yes", :value => true %> </div>
```
Comment (RathanakSreang) [inline app/views/supervisor/courses/new.html.erb]: `value: true`

File: app/views/shared/_error_messages.html.erb [hunk 12]
```diff
@@ -1,8 +1,8 @@
 <% if object.errors.any? %>
   <div id="error_explanation">
     <div class="alert alert-danger">
-      The form contains <%= pluralize(object.errors.count, "error") %>.
-   </div>
+      The form contains <%= pluralize(object.errors.count, "error") %>
```
Comment (HorNorin) [inline app/views/shared/_error_messages.html.erb]: remove `(`

File: app/views/supervisor/courses/new.html.erb [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    <%= form_for [:supervisor, @course] do |f| %>
+      <%= render "shared/error_messages", object: f.object %>
+      <%= f.label :name %>
+      <%= f.text_field :name, class: "form-control" %>
+      <%= f.label :description %>
+      <%= f.text_area :description, class: "form-control" %>
+      <div class="container-fluid">
+       <%= f.label :status %>
+       <div class="row">
+        <div class="col-md-1"> <%= f.label :status, "Yes", :value true %> </div>
```
Comment (RathanakSreang) [inline app/views/supervisor/courses/new.html.erb]: `value: true`

File: app/views/supervisor/courses/new.html.erb [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+        <div class="col-md-1"> <%= f.radio_button :status, "start" %> </div>
+      </div>
+      <div class="row">
+       <div class="col-md-1"> <%= f.label :status, "No", :value false %> </div>
```
Comment (RathanakSreang) [inline app/views/supervisor/courses/new.html.erb]: `value: false`

File: config/routes.rb [hunk 15]
```diff
@@ -6,5 +6,6 @@
   resources :users
   namespace :supervisor do
```
Comment (dieunb) [inline config/routes.rb]: root page cùa supervisor namespace?

File: app/controllers/supervisor/courses_controller.rb [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+    @course = Course.new course_params
+    if @course.save
+      flash[:info] = "Create Successfull"
+      redirect_to root_url
+    else
+      render "new"
+    end
+  end
+
+  def show
```
Comment (dieunb) [inline app/controllers/supervisor/courses_controller.rb]: Hàm này chưa xử lý thì xóa đi em.

File: app/controllers/supervisor/courses_controller.rb [hunk 17]
```diff
@@ -0,0 +1,20 @@
+class Supervisor::CoursesController < ApplicationController
+  def new
+    @course = Course.new
+  end
+  
+  def create
+    @course = Course.new course_params
+    if @course.save
+      flash[:info] = "Create Successfull"
```
Comment (dieunb) [inline app/controllers/supervisor/courses_controller.rb]: i18n

File: app/views/supervisor/courses/new.html.erb [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+    <%= form_for [:supervisor, @course] do |f| %>
+      <%= render "shared/error_messages", object: f.object %>
+      <%= f.label :name %>
+      <%= f.text_field :name, class: "form-control" %>
+      <%= f.label :description %>
+      <%= f.text_area :description, class: "form-control" %>
+      <div class="container-fluid">
+       <%= f.label :status %>
+       <div class="row">
+        <div class="col-md-1"> <%= f.label :status, "Yes", value: true %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: i18n

File: app/views/supervisor/courses/new.html.erb [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+        <div class="col-md-1"> <%= f.radio_button :status, "start" %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: i18n

File: app/views/supervisor/courses/new.html.erb [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+      </div>
+      <div class="row">
+       <div class="col-md-1"> <%= f.label :status, "No", value: false %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: i18n

File: app/views/supervisor/courses/new.html.erb [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+       <div class="col-md-1"> <%= f.radio_button :status, "not start", checked: true %> </div>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: i18n

File: app/views/supervisor/courses/new.html.erb [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+      </div>
+      </div>
+      <%= f.submit "Create Course", class: "btn btn-primary" %>
```
Comment (dieunb) [inline app/views/supervisor/courses/new.html.erb]: i18n

**Pull request merged**